src/sentry/api/endpoints/organization_events_meta.py

```diff
@@ -88,6 +88,94 @@ def get(self, request, organization):
+    def get(self, request, organization, trace_id):
+        if not self.has_feature(organization, request):
+            return Response(status=404)
+
+        try:
+            params = self.get_snuba_params(request, organization)
+        except NoProjects:
+            return Response(status=404)
+
+        # Get 100 events from snuba, and pray these cover the trace
+        with self.handle_query_errors():
+            result = discover.query(
+                selected_columns=[
+                    "id",
+                    "timestamp",
+                    "transaction",
+                    "project_id",
+                    "trace.span",
+                    "trace.parent_span",
+                ],
+                orderby=["-timestamp", "trace.parent_span", "id"],
+                params=params,
+                query=f"event.type:transaction trace:{trace_id}",
+                limit=100,
+                referrer="experimental.api.trace-view.get_ids",
+            )
+            if len(result["data"]) == 0:
+                return Response(status=404)
+
+        # Get all our events
+        events = {
+            event.event_id: event
+            for event in eventstore.get_events(
+                eventstore.Filter(
+                    project_ids=params["project_id"],
+                    event_ids=[event["id"] for event in result["data"]],
+                )
+            )
+        }
+
+        with sentry_sdk.start_span(op="parsing snuba response"):
+            # A potential root is the transaction with no parent_span
+            # TODO: Not 100% accurate since its possilbe to have orphans in a trace.
+            root_data = [item for item in result["data"] if item["trace.parent_span"] == ""][0]
+            root = events[root_data["id"]]
+            root_span_id = root_data["trace.span"]
+
+            trace = {}
+            trace[root_span_id] = {span["span_id"]: [] for span in root.data.get("spans", [])}
+
+            # TODO these mappings should either all be span ids or parent span ids, currently an ugly mix of both
+            project_map = {item["trace.parent_span"]: item for item in result["data"]}
+            project_map.update({item["trace.span"]: item for item in result["data"]})
+            transaction_map = {item["trace.span"]: item["transaction"] for item in result["data"]}
+            transaction_map.update(
+                {item["trace.parent_span"]: item["transaction"] for item in result["data"]}
+            )
+
+        with sentry_sdk.start_span(op="fill_children"):
+            self.fill_children(root_span_id, project_map, events, trace)
+
+        return Response(self.map_transaction(trace, transaction_map))
+
+    def fill_children(self, span_id, project_map, events, trace):
+        to_delete = []
+
+        for child_id in trace[span_id].keys():
+            if child_id in project_map:
+                child = project_map[child_id]
+                child_event = events[child["id"]]
```

Comment:
Correct me if I'm wrong but `child_id` and `child["id"]` are different things here right? If so, the naming needs work here.

Reply:
🙃 yeah, if there's another iteration I'll rename `child_id` to `child_span_id`